server/conf/application.conf

```diff
@@ -611,3 +611,16 @@ file_upload_allowed_file_type_specifiers = ${?FILE_UPLOAD_ALLOWED_FILE_TYPE_SPEC
 # The number of minutes before a session expires
 maximum_session_duration_minutes = 600
 maximum_session_duration_minutes = ${?MAXIMUM_SESSION_DURATION_MINUTES}
+
+# Session timeout settings
+session_timeout_enabled = false
+session_timeout_enabled = ${?SESSION_TIMEOUT_ENABLED}
+
+session_inactivity_timeout_minutes = 30
+session_inactivity_timeout_minutes = ${?SESSION_INACTIVITY_TIMEOUT_MINUTES}
+
+session_inactivity_warning_minutes = 5  # 5 minutes warning for inactivity
+session_inactivity_warning_minutes = ${?SESSION_INACTIVITY_WARNING_MINUTES}
+
+session_max_duration_warning_minutes = 10     # 10 minutes warning for total duration
```

Comment:
question: Does this warn the user 10 mins _before_ the total duration? If so, can you make that clear here?

Reply:
(question) How does `session_inactivity_timeout_minutes` differ from `maximum_session_duration_minutes` which is already defined?

Reply:
Danielle asked the same thing and amit answered it here https://github.com/civiform/civiform/pull/9803#discussion_r1973747373

This is making me think we might need something more in the description to make that clearer.

Reply:
I think we should try to unify the naming on all of these. fyi @dkatzz 

- maximum_session_duration_minutes
- session_timeout_enabled
- session_inactivity_timeout_minutes
- session_inactivity_warning_minutes
- session_max_duration_warning_minutes


or at least make `session_max_duration_warning_minutes` -> `session_duration_warning_threshold_minutes`  or something like that. 

It's hard to follow since things are so similarly named.

Reply:
Yes. that is correct. I updated the comment.

Reply:
The pattern I following is 
For warning: session_<reason>_warning_minutes
For timeout: session_<reason>_timeout_minutes
the reason can be either inactivity or max_duration.
@gwendolyngoetz Love to get your thoughts on how to unify this.

Reply:
How about this?
session_timeout_enabled
maximum_session_duration_minutes(existing)
session_inactivity_duration_minutes
session_duration_warning_threshold_minutes
session_inactivity_warning_threshold_minutes

Reply:
Updated the naming to this:
```
session_timeout_enabled
maximum_session_duration_minutes(existing)
session_inactivity_timeout_minutes
session_duration_warning_threshold_minutes
session_inactivity_warning_threshold_minutes
```

Reply:
This is much more clear to me. I'm ready to approve but want to hear if this works for @gwendolyngoetz and @dkatzz first.

Reply:
Agreed this is much more clear!

My only thought is whether we should change 'session_inactivity_timeout_minutes' to 'maximum_session_inactivity_timeout_minutes' for consistency with the existing on or update 'maximum_session_duration_minutes' to 'session_duration_timeout_minutes' but would need to ensure nobody is setting that so we don't cause issues with renaming the variable. If we renamed that would be a separate PR. 

Curious what others think!

Reply:
I think we `maximum_session_duration_minutes` to `session_duration_timeout_minutes` makes more sense to me. I can make that change in this PR or we can change it later.
 
The problem with `maximum_session_inactivity_timeout_minutes` is that prefix `maximum` is confusing. It is not maximum of the session inactivity timeout but rather It is just a timeout.

Reply:
Sounds good! I'm good with this but we'll just want to check nobody is setting the other flag before we make the other change

Reply:
that makes sense to me!

Reply:
Yeah looks good. Thanks!